frontend/src/api/Socket.ts

```diff
@@ -3,14 +3,52 @@ import Stomp from 'stompjs';
 
 let stompClient:any = null;
 
-export const connect = (endpoint:string) => {
-  const socket:any = new SockJS(endpoint);
-  stompClient = Stomp.over(socket);
-  stompClient.connect();
+type User = {
+  nickname: string;
+}
+
+// Variable to hold the current connected state.
+let connected = false;
+
+// Create constants for the subscription and send message URLs.
+const SUBSCRIBE_URL:string = '/api/v1/socket/subscribe-user';
+const SEND_USER_URL:string = '/api/v1/socket/user';
+
+export const sendUser = (nickname:string) => {
+  if (connected) {
+    const nicknameInput:string = (nickname !== '') ? nickname : 'Anonymous';
+    stompClient.send(SEND_USER_URL, {}, nicknameInput);
+  } else {
+    console.error('You must be connected to a socket before sending user information.');
+  }
+};
+
+export const connect = (endpoint:string, nickname:string) => {
+  if (!connected) {
+    // Connect to given endpoint, subscribe to future messages, and send user message.
+    const socket:any = new SockJS(endpoint);
+    stompClient = Stomp.over(socket);
+    stompClient.connect({}, () => {
+      stompClient.subscribe(SUBSCRIBE_URL, (user:any) => {
+        const userObject:User = JSON.parse(user.body);
```

Comment:
Question: Is it possible to get stomp to return `user` as an object rather than a string?

Reply:
In the line `stompClient.subscribe(SUBSCRIBE_URL, (user:any) => {`, `user` **is** an object; however, this object is generated by STOMP protocol. You may be referring to `user.body`, which **is** a string; as far as I can tell, the `user` object is generated by the STOMP protocol, and it places the string JSON of the object send from the backend into `user.body` -- so I'm not sure if it is possible to return this as an object. Is there a reason you'd want this to be an object (is it just so we don't have to have the `JSON.parse(user.body)` method, or is there another reason)?

Here is a screenshot of the `user` object in the console with the nickname `"Anonymous"`:
<img width="364" alt="User Object in Console" src="https://user-images.githubusercontent.com/7987279/92681497-a13ece80-f2e2-11ea-9b74-913e4b5087c1.png">

I'd also note that this is the format that the Spring tutorial used, so I am basing this off of something.